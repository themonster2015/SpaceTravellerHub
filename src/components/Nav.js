import React from 'react';
import { NavLink } from 'react-router-dom';

const Nav = () => (
  <nav>
    <div className="nav-items">
      <NavLink to="/" activeClassName="selected" exact>Rockets</NavLink>
      <NavLink to="/missions" activeClassName="selected">Missions</NavLink>
      <NavLink to="/profile" activeClassName="selected">My Profile</NavLink>
    </div>
  </nav>
);

export default Nav;
