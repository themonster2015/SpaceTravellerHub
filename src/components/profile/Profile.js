import React from 'react';
import { useSelector } from 'react-redux';

const Profile = () => {
  const rockets = useSelector((state) => state.rocketsReducer.rockets);
  const missions = useSelector((state) => state.missions);

  const reservedRockets = rockets.filter((rocket) => rocket.reserved);
  const reservedMissions = missions.filter((mission) => mission.reserved);

  return (
    <div className="main-container">
      <div className="missions-section">
        <h2>My Missions</h2>
        {
          reservedMissions.map((mission) => (
            <div key={mission.mission_id} className="reserved">{mission.mission_name}</div>
          ))
        }
      </div>
      <div className="rockets-section">
        <h2>My Rockets</h2>
        {
          reservedRockets.map((rocket) => (
            <div key={rocket.id} className="reserved">{rocket.rocket_name}</div>
          ))
        }
      </div>
    </div>
  );
};

export default Profile;
