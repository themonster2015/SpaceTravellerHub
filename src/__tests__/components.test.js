import React from 'react';
import renderer from 'react-test-renderer';
import { BrowserRouter as Router } from 'react-router-dom';
import { Provider } from 'react-redux';
import Profile from '../components/profile/Profile';
import Nav from '../components/Nav';
import store from '../redux/configureStore';

describe('Test that all components are rendering just fine.', () => {
  it('Render the Nav', () => {
    const tree = renderer
      .create(<Router><Nav /></Router>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Render the Profile Page', () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <Profile />
        </Provider>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
