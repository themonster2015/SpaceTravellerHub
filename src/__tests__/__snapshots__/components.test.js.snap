// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test that all components are rendering just fine. Render the Missions page. 1`] = `
<div
  className="container"
>
  <table>
    <thead>
      <tr>
        <th
          className="mTitle"
        >
          Mission
        </th>
        <th
          className="mDes"
        >
          Description
        </th>
        <th
          className="mStatus"
        >
          Status
        </th>
        <th>
           
        </th>
      </tr>
    </thead>
    <tbody />
  </table>
</div>
`;

exports[`Test that all components are rendering just fine. Render the Nav 1`] = `
<nav>
  <div
    className="nav-items"
  >
    <a
      aria-current="page"
      className="selected"
      href="/"
      onClick={[Function]}
      style={Object {}}
    >
      Rockets
    </a>
    <a
      aria-current={null}
      href="/missions"
      onClick={[Function]}
    >
      Missions
    </a>
    <a
      aria-current={null}
      href="/profile"
      onClick={[Function]}
    >
      My Profile
    </a>
  </div>
</nav>
`;

exports[`Test that all components are rendering just fine. Render the Profile Page 1`] = `
<div
  className="main-container"
>
  <div
    className="missions-section"
  >
    <h2>
      My Missions
    </h2>
  </div>
  <div
    className="rockets-section"
  >
    <h2>
      My Rockets
    </h2>
  </div>
</div>
`;

exports[`Test that all components are rendering just fine. Render the Rockets Page. 1`] = `null`;
